bookkeeper-server/src/main/java/org/apache/bookkeeper/client/SyncCallbackUtils.java

```diff
@@ -0,0 +1,287 @@
+
+        private final CompletableFuture future;
+
+        public SyncCreateCallback(CompletableFuture future) {
+            this.future = future;
+        }
+
+        /**
+         * Create callback implementation for synchronous create call.
+         *
+         * @param rc return code
+         * @param lh ledger handle object
+         * @param ctx optional control object
+         */
+        @Override
+        @SuppressWarnings(value = "unchecked")
+        public void createComplete(int rc, LedgerHandle lh, Object ctx) {
+            finish(rc, lh, future);
+        }
+
+    }
+
+    static class SyncOpenCallback implements AsyncCallback.OpenCallback {
+
+        private final CompletableFuture future;
+
+        public SyncOpenCallback(CompletableFuture future) {
+            this.future = future;
+        }
+
+        /**
+         * Callback method for synchronous open operation
+         *
+         * @param rc
+         *          return code
+         * @param lh
+         *          ledger handle
+         * @param ctx
+         *          optional control object
+         */
+        @Override
+        @SuppressWarnings("unchecked")
+        public void openComplete(int rc, LedgerHandle lh, Object ctx) {
+            finish(rc, lh, future);
+        }
+    }
+
+    static class SyncDeleteCallback implements AsyncCallback.DeleteCallback {
+
+        private final CompletableFuture<Void> future;
+
+        public SyncDeleteCallback(CompletableFuture<Void> future) {
+            this.future = future;
+        }
+
+
+        /**
+         * Delete callback implementation for synchronous delete call.
+         *
+         * @param rc
+         *            return code
+         * @param ctx
+         *            optional control object
+         */
+        @Override
+        @SuppressWarnings("unchecked")
+        public void deleteComplete(int rc, Object ctx) {
+            finish(rc, null, future);
+        }
+    }
+
+    static class LastAddConfirmedCallback implements AsyncCallback.AddLacCallback {
+        static final LastAddConfirmedCallback INSTANCE = new LastAddConfirmedCallback();
+        /**
+         * Implementation of callback interface for synchronous read method.
+         *
+         * @param rc
+         *          return code
+         * @param lh
+         *          ledger identifier
+         * @param ctx
+         *          control object
+         */
+        @Override
+        public void addLacComplete(int rc, LedgerHandle lh, Object ctx) {
+            if (rc != BKException.Code.OK) {
+                LOG.warn("LastAddConfirmedUpdate failed: {} ", BKException.getMessage(rc));
+            } else {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Callback LAC Updated for: {} ", lh.getId());
+                }
+            }
+        }
+    }
+
+    static class SyncReadCallback implements AsyncCallback.ReadCallback {
```

Comment:
why do we have the old style callback here? can you make this one consistent with SyncOpenCallback?

Reply:
done